### PR TITLE
add support for OpenBSD

### PR DESCRIPTION
**Pull request by qbit**
This PR lets OpenRW build on OpenBSD! Game starts and is playable!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,10 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 	add_definitions(-DRW_OSX)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
 	add_definitions(-DRW_FREEBSD)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+	add_definitions(-DRW_OPENBSD)
+	include_directories("/usr/X11R6/include")
+	set(OPENRW_PLATFORM_LIBS iconv)
```
Comment (JayFoxRox) [inline CMakeLists.txt]: We depend on iconv on all platforms but will get rid of it soon (so this hack is fine).

File: CMakeLists.txt [hunk 2]
```diff
@@ -36,6 +36,10 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 	add_definitions(-DRW_OSX)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
 	add_definitions(-DRW_FREEBSD)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
+	add_definitions(-DRW_OPENBSD)
+	include_directories("/usr/X11R6/include")
```
Comment (JayFoxRox) [inline CMakeLists.txt]: This should probably be changed to some CMake magic.

**Comment by JayFoxRox** [comment]
This might be an absolutely stupid question, but we support FreeBSD already, wouldn't compiling for that also work for OpenBSD? (e.g. can't we just call it RW_BSD?)

File: CMakeLists.txt (hunk 2 above)
Comment (danhedron) [inline CMakeLists.txt]: Is this explicit path required? Can cmake find it for us?

**Comment by mulander** [comment]
@JayFoxRox FreeBSD and OpenBSD are different kernels, user lands and available third party Libraries. While they do share some common ancestry code they are different systems with different feature sets. It might compile now but could soon become problematic with one OS lacking features of the other or implementing them in a vastly different way.

File: CMakeLists.txt (hunk 2 above)
Comment (mulander) [inline CMakeLists.txt]: This path is constant on OpenBSD. [Xenocara](https://en.wikipedia.org/wiki/Xenocara) is OpenBSD's version of the x.org server. It won't move around on this platform :)

Comment (krytarowski) [inline CMakeLists.txt]: Not really, it can be dynamic with pkgsrc for OpenBSD. Hardcoding this sort of things makes more harm than good.

**Comment by krytarowski** [comment]
After merging it I will push NetBSD support.

**Comment by JayFoxRox** [comment]
PC did not explode, also did not forcefully download OpenBSD and shoved it on my harddrive.
Still works on Arch Linux.

Code review process was very hard and long but also passed:

LGTM

PS: I want at least one more LGTM from an OpenBSD user on this, possibly also MacOS or even Windows, I've already mentioned that on IRC.

**Comment by krytarowski** [comment]
OK from NetBSD user :)

**Comment by christoph-heiss** [comment]
+1 for macOS.

**Comment by mulander** [comment]
Waiting for GTA 3 game data to download before doing a test run on OpenBSD -current amd64

File: CMakeLists.txt [hunk 3]
```diff
@@ -36,6 +36,9 @@ elseif (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
 	add_definitions(-DRW_OSX)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
 	add_definitions(-DRW_FREEBSD)
+elseif(CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
```
Comment (JayFoxRox) [inline CMakeLists.txt]: nit: Commit name "OpenBSD support" or something like that would have been better (to suggest that OpenBSD wasn't supported previously). "Fixes" usually implies something was existing + broken.

**Comment by mulander** [comment]
Tested on OpenBSD/amd64 -current

The game builds without issues and can be played. LGTM from an OpenBSD user :)

```
$ ./rwgame/rwgame
I [Game] Game directory: /home/mulander/Games/GTA3/
I [Game] Build: a8db29cd
Max UBO Size: 65536
UBO Alignment: 16
Max batch size: 512
I [Renderer] OpenGL Renderer Version: 3.3 (Core Profile) Mesa 11.2.2 (GLSL 3.30) Vendor: Intel Open Source Technology Center
I [Data] Loaded 42 zones from /home/mulander/Games/GTA3/data/gta3.zon
I [Game] Started
Error opening sound file "/home/mulander/Games/GTA3//audio/City.wav": System error : No such file or directory.
I [World] Loaded cutscene: BET
I [World] Loaded cutscene: JB
I [World] Creating Vehicle ID 116 (POLICAR)
I [World] Creating Vehicle ID 117 (ENFORCR)
I [World] Creating Vehicle ID 130 (RUMPO)
I [World] Creating Vehicle ID 96 (PATRIOT)
I [World] Creating Vehicle ID 110 (TAXI)
I [World] Creating Vehicle ID 105 (CHEETAH)
I [World] Creating Vehicle ID 92 (STINGER)
I [World] Creating Vehicle ID 105 (CHEETAH)
I [World] Creating Vehicle ID 92 (STINGER)
I [World] Creating Vehicle ID 111 (KURUMA)
I [World] Creating Vehicle ID 110 (TAXI)
I [World] Creating Vehicle ID 111 (KURUMA)
I [World] Creating Vehicle ID 110 (TAXI)
I [World] Creating Vehicle ID 111 (KURUMA)
I [World] Creating Vehicle ID 111 (KURUMA)
I [World] Creating Vehicle ID 111 (KURUMA)
I [World] Creating Vehicle ID 93 (LINERUN)
I [World] Creating Vehicle ID 108 (MOONBM)
I [World] Creating Vehicle ID 116 (POLICAR)
I [World] Creating Vehicle ID 116 (POLICAR)
I [World] Creating Vehicle ID 91 (IDAHO)
I [World] Creating Vehicle ID 130 (RUMPO)
I [World] Creating Vehicle ID 91 (IDAHO)
I [World] Creating Vehicle ID 109 (ESPERAN)
I [Game] Beginning cleanup
I [Game] Stopping work queue
I [Game] Cleaning up scripts
I [Game] Cleaning up renderer
I [Game] Cleaning up world
I [Game] Cleaning up state
I [Game] Cleaning up window
I [Game] Cleaning up work queue
I [Game] Done cleaning up
```

**Comment by danhedron** [comment]
+1 from me.

**Comment by brightersun99** [comment]
glad to see this never got anywhere

**Pull request merged**